### ISSUE TITLE
viewport: wrapped lines keep style until resized

### ISSUE THREAD

**Issue by bevicted** [charmbracelet/bubbles#681]
**Describe the bug**
Styling seems to get "stuck" for wrapped lines in viewport. This gets corrected every time I resize the window.

Note that whatever happens, the first and last lines work as intended.

As a little extra: I tried what happens with `HighPerformanceRendering` set to `true`, but nothing seems to get rendered.

**Setup**
Please complete the following information along with version numbers, if applicable.
 - OS: MacOS
 - Shell: zsh
 - Terminal Emulator: alacritty
 - Terminal Multiplexer: tmux
 - Locale: en_US.UTF-8

**To Reproduce**
1. create viewport with a list of long strings, a cursor and highlight styling
2. initialise
3. move cursor

**Source Code**
Relevant code parts (cut some logic out of it which doesn't change styling/rendering):

```go
func (li *LogInspector) Update(msg tea.Msg) (*LogInspector, tea.Cmd) {
  // logic to move cursor with keys
}

func (li *LogInspector) View() string {
        // note: same result without applying a width style here
	return lipgloss.NewStyle().Width(li.width).Render(li.viewport.View())
}

func (li *LogInspector) updateViewport() {
	var sb strings.Builder

	for i := range li.logs {
		sb.WriteString(li.renderLog(i))
	}

	li.viewport.SetContent(sb.String())
}


func (li *LogInspector) renderLog(idx int) string {
	logLine := li.logs[idx]

	if idx == li.cursor {
		logLine = highlightStyle.Render(logLine)
	}

	return logLine + "\n"
}

```

**Expected behavior**
Styling to change.

**Screenshots**
initially correct:
<img width="904" alt="image" src="https://github.com/user-attachments/assets/8594d79a-ea33-4c77-b48d-6a802e3fabce">

after moving the cursor one down:
<img width="908" alt="image" src="https://github.com/user-attachments/assets/4e9a2c9c-9f49-4380-93f9-d2fbd6cccb1c">

after resizing my window:
<img width="910" alt="image" src="https://github.com/user-attachments/assets/b9620515-4f1d-4047-9f6d-91b610e00c10">

**Additional context**
N/A

**Comment by bashbunni** [comment]
It looks like in the original example you're styling the text before wrapping it. I would expect that to cause some issues with the highlighted style not reaching the right rows.

@bevicted do you think this is a viewport bug or are you satisfied with your fix? If it's a bug, it would be really helpful to get a minimal reproducible example from you so we can try and debug on our end :)

**Comment by bevicted** [comment]
Hey @bashbunni

> I would expect that to cause some issues with the highlighted style not reaching the right rows.

My main issue here is when I create the viewport, I set the dimensions via `viewport.New(w, h)`, upon resize I update these fields directly `viewport.Width = width`, yet I encounter this issue if I do not set the width of the strings contained within, via lipgloss. I would assume that since I set the size for the viewport and it automatically wraps my lines, that styling would work on wrapped lines.

Now I set every width directly and in addition carry a `widthStyle` lipgloss style on my struct which is yet another size to update and apply to everything.

I guess my problem is twofold:

- `m.Width` `m.Height` vs `style.Width()` `style.Height()` is unclear to me, I would like to stick with one but I found that sometimes I need to experiment to decide which one I need or in this case, both.
- viewport is trying to do a lot here but only goes halfway
    - either go full in and try to handle styles when wrapping lines
    - or just drop auto line wrapping and leave it up to the user to implement it

I would honestly prefer if auto line wrapping would be just dropped as it's literally one line to do with lipgloss, and would be a lot easier on viewport's code, however I would expect a note on `style.Width()` that wrapping may break styling that's applied before it. (Which viewport does automatically)

**Comment by bevicted** [comment]
Coming back to this, wrapping is causing me even more problems, checking the code, it seems to me that the only way to have a bigger width is to set `viewport.Width` to a big number (not too big, as it panics with `math.MaxInt`) and unsetting `viewport.Style`'s width and maxwidth.
However I was still unable to make it work, it kept wrapping.

**Comment by meowgorithm** [comment]
Apologies about the trouble this is causing you, @bevicted. What you'll want to do is listen for window sizes and re-style and re-wrap on resize. I've included a simple example showing how to do this below.

Also, side note that performance rendering is deprecated: Bubble Tea's standard renderer has improved quite a lot since we introduced it + we have an even more efficient renderer on the way.

```go
package main

import (
	"github.com/charmbracelet/bubbles/viewport"
	tea "github.com/charmbracelet/bubbletea"
	"github.com/charmbracelet/lipgloss"
)

type model struct {
	vp viewport.Model
}

func (m model) Init() tea.Cmd {
	return nil
}

func (m model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
	switch msg := msg.(type) {

	case tea.KeyMsg:
		switch msg.String() {
		case "q", "esc", "ctrl+c":
			return m, tea.Quit
		}

	case tea.WindowSizeMsg:
		m.vp.Width = msg.Width
		m.vp.Height = msg.Height

		s := lipgloss.NewStyle().
			Width(m.vp.Width).
			Foreground(lipgloss.Color("63")).
			Render(content)
		m.vp.SetContent(s)
		return m, nil
	}

	m.vp, _ = m.vp.Update(msg)
	return m, nil
}

func (m model) View() string {
	return m.vp.View()
}

func main() {
	m := model{vp: viewport.New(0, 0)}
	p := tea.NewProgram(m)
	if _, err := p.Run(); err != nil {
		panic(err)
	}
}

const content = `I stuffed a shirt or two into my old carpet-bag, tucked it under my arm, and started for Cape Horn and the Pacific.  Quitting the good city of old Manhatto, I duly arrived in New Bedford.  It was a Saturday night in December.  Much was I disappointed upon learning that the little packet for Nantucket had already sailed, and that no way of reaching that place would offer, till the following Monday.

As most young candidates for the pains and penalties of whaling stop at this same New Bedford, thence to embark on their voyage, it may as well be related that I, for one, had no idea of so doing.  For my mind was made up to sail in no other than a Nantucket craft, because there was a fine, boisterous something about everything connected with that famous old island, which amazingly pleased me.  Besides though New Bedford has of late been gradually monopolising the business of whaling, and though in this matter poor old Nantucket is now much behind her, yet Nantucket was her great original--the Tyre of this Carthage;--the place where the first dead American whale was stranded.  Where else but from Nantucket did those aboriginal whalemen, the Red-Men, first sally out in canoes to give chase to the Leviathan?  And where but from Nantucket, too, did that first adventurous little sloop put forth, partly laden with imported cobblestones--so goes the story--to throw at the whales, in order to discover when they were nigh enough to risk a harpoon from the bowsprit?

Now having a night, a day, and still another night following before me in New Bedford, ere I could embark for my destined port, it became a matter of concernment where I was to eat and sleep meanwhile.  It was a very dubious-looking, nay, a very dark and dismal night, bitingly cold and cheerless.  I knew no one in the place.  With anxious grapnels I had sounded my pocket, and only brought up a few pieces of silver,--So, wherever you go, Ishmael, said I to myself, as I stood in the middle of a dreary street shouldering my bag, and comparing the gloom towards the north with the darkness towards the south--wherever in your wisdom you may conclude to lodge for the night, my dear Ishmael, be sure to inquire the price, and don't be too particular.

With halting steps I paced the streets, and passed the sign of "The Crossed Harpoons"--but it looked too expensive and jolly there. Further on, from the bright red windows of the "Sword-Fish Inn," there came such fervent rays, that it seemed to have melted the packed snow and ice from before the house, for everywhere else the congealed frost lay ten inches thick in a hard, asphaltic pavement,--rather weary for me, when I struck my foot against the flinty projections, because from hard, remorseless service the soles of my boots were in a most miserable plight.  Too expensive and jolly, again thought I, pausing one moment to watch the broad glare in the street, and hear the sounds of the tinkling glasses within. But go on, Ishmael, said I at last; don't you hear? get away from before the door; your patched boots are stopping the way.  So on I went.  I now by instinct followed the streets that took me waterward, for there, doubtless, were the cheapest, if not the cheeriest inns.`
```

**Comment by bevicted** [comment]
Hey @meowgorithm,

After I posted yesterday, I removed the viewport from my code and just listed things only to find out that my strings are still wrapped.
I then realized that the way I implemented tabs is to have a main tea.Model that's View will render a `header + activeTab.View() + footer` and I applied a style with `Width` and `MaxWidth` to all of these, then have trouble in my tab's `View`.

Entirely skill issue on my part, sorry for wasting your time.

Your example shows me however that I might be doing it wrong, calculating and styling the final render in `View`, though I saw it being done this way in others' projects.
